internal/config/watchd/watchd_test.go

```diff
@@ -0,0 +1,209 @@
+
+		want    string
+		wantErr bool
+	}{
+		"empty args":                        {wantErr: true},
+		"no config argument":                {args: "adwatchd.exe", wantErr: true},
+		"short config argument":             {args: `adwatchd.exe -c C:\path\to\adwatchd.yml`, want: `C:\path\to\adwatchd.yml`},
+		"short config argument with quotes": {args: `adwatchd.exe -c "C:\path\to\adwatchd.yml"`, want: `C:\path\to\adwatchd.yml`},
+	}
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			got, err := watchdconfig.GetConfigFileFromArgs(tc.args)
+			if tc.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+			require.Equal(t, tc.want, got)
+		})
+	}
+}
+
+func TestGetDirsFromConfigFile(t *testing.T) {
+	tests := map[string]struct {
+		emptyConfig bool
+		noConfig    bool
+		noDirs      bool
+		badDirs     bool
+
+		wantDirs []string
+	}{
+		"no config file":              {noConfig: true},
+		"empty config file":           {emptyConfig: true},
+		"no dirs in config file":      {noDirs: true},
+		"config dirs is not an array": {badDirs: true},
+		"config dirs is an array":     {wantDirs: []string{"/path/to/dir1", "/path/to/dir2"}},
+	}
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			var data []byte
+			var err error
+
+			goldPath := filepath.Join("testdata", "golden", strings.Replace(name, " ", "_", -1))
+			if update {
+				appConfig := watchdconfig.AppConfig{Dirs: tc.wantDirs}
+
+				// Handle error cases
+				if tc.noConfig {
+					// no config file
+				} else if tc.emptyConfig {
+					_, err = os.Create(goldPath)
+					require.NoError(t, err, "failed to create empty config file")
+				} else if tc.noDirs {
+					appConfig.Dirs = nil
+					data, err = yaml.Marshal(&appConfig)
+					require.NoError(t, err, "failed to marshal config")
+					err = os.WriteFile(goldPath, data, 0600)
+					require.NoError(t, err, "failed to write config")
+				} else if tc.badDirs {
+					err = os.WriteFile(goldPath, []byte(`- dirs: "testdir"`), 0600)
+					require.NoError(t, err, "failed to write config")
+				} else {
+					// Normal case
+					data, err = yaml.Marshal(&appConfig)
+					require.NoError(t, err, "failed to marshal config")
+					err = os.WriteFile(goldPath, data, 0600)
+					require.NoError(t, err, "failed to write config")
+				}
+			}
+			got := watchdconfig.GetDirsFromConfigFile(goldPath)
+			require.ElementsMatch(t, tc.wantDirs, got)
+		})
+	}
+}
+
+func TestFilterAbsentDirs(t *testing.T) {
+	tests := map[string]struct {
+		inputDirs    []string
+		existingDirs []string
+
+		wantDirs []string
+	}{
+		"no existing dirs":   {inputDirs: []string{"dir1", "dir2"}},
+		"some existing dirs": {inputDirs: []string{"dir1", "dir2"}, existingDirs: []string{"dir1"}, wantDirs: []string{"dir1"}},
+		"all existing dirs":  {inputDirs: []string{"dir1", "dir2"}, existingDirs: []string{"dir1", "dir2"}, wantDirs: []string{"dir1", "dir2"}},
```

Comment:
In your code, you check that the dest is a dir. I suggest that you have a case where it’s a file.

Reply:
Sorry, I was probably unclear, but I meant an additional test (one where both are dirs, as you had, one or more tests where one is a file, and maybe a mix of both?)

Reply:
No more tests for this as no more functions :)